src/runtime/thread_pool_common.h

```diff
@@ -239,6 +239,9 @@ using namespace Halide::Runtime::Internal;
 extern "C" {
 
 WEAK int halide_set_num_threads(int n) {
```

Comment:
Hmm, good question, actually: I had looked at the code around line 157 onwards and thought that desired_num_threads == 0 means "use env vars, etc"... but looking again I realize that applies only if the work queue isn't yet initialized. I'll pull that code out into a helper function.